cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -140,11 +154,90 @@ impl ClientTrait for CatalogClient {
                 api_search_result
                     .total_count
                     .try_into()
-                    .map_err(|_| SearchError::NegativeNumberOfResults)?,
+                    .map_err(|_| CatalogClientError::NegativeNumberOfResults)?,
             ),
         };
         Ok(search_results)
     }
+
+    /// Wrapper around the autogenerated
+    /// [catalog_api_v1::Client::packages_api_v1_catalog_packages_pkgpath_get]
+    async fn package_versions(
+        &self,
+        attr_path: impl AsRef<str> + Send + Sync,
+    ) -> Result<SearchResults, VersionsError> {
+        let attr_path = attr_path.as_ref();
+        let stream = make_depaging_stream(
+            |page_number, page_size| async move {
+                let response = self
+                    .client
+                    .packages_api_v1_catalog_packages_pkgpath_get(
+                        attr_path,
+                        Some(page_number),
+                        Some(page_size),
+                    )
+                    .await
+                    .map_err(|e| {
+                        if CatalogClientError::is_unexpected_error(&e) {
+                            CatalogClientError::UnexpectedError(e).into()
+                        } else {
+                            VersionsError::Versions(e)
+                        }
+                    })?;
+
+                let packages = response.into_inner();
+
+                Ok::<_, VersionsError>((
+                    packages.total_count,
+                    packages
+                        .items
+                        .into_iter()
+                        .map(TryInto::<SearchResult>::try_into)
+                        .collect::<Result<Vec<_>, _>>()?,
+                ))
+            },
+            10,
+        );
+
+        let results: Vec<SearchResult> = stream.try_collect().await?;
+        let count = Some(results.len() as u64);
+        Ok(SearchResults { results, count })
+    }
+}
+
+/// Take a function that takes a page_number and page_size and returns a
+/// total_count of results and a Vec of results on a page.
+///
+/// Create a stream that yields all results from all pages.
+fn make_depaging_stream<T, E, Fut>(
+    generator: impl Fn(i64, i64) -> Fut,
+    page_size: i64,
```

Comment:
If this will loop forever when `page_size` is zero, then we can take a `NonZeroI32` parameter instead of `i32`.